packages/block-library/src/quote/transforms.js

```diff
@@ -75,43 +129,116 @@ const transforms = {
 					node.nodeName === 'BLOCKQUOTE' &&
 					// The quote block can only handle multiline paragraph
 					// content with an optional cite child.
-					Array.from( node.childNodes ).every(
-						isParagraphOrSingleCite
-					)
+					Array.from( node.childNodes ).every( isAllowedNode )
+				);
+			},
+			transform: ( node ) => {
+				const innerBlocks = [];
+				let cite = '';
+				node.childNodes.forEach( ( childNode ) => {
+					switch ( childNode.nodeName ) {
+						case 'P':
+							innerBlocks.push(
+								createBlock( 'core/paragraph', {
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H1':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 1,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H2':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 2,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H3':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 3,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H4':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 4,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H5':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 5,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'H6':
+							innerBlocks.push(
+								createBlock( 'core/heading', {
+									level: 6,
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'UL':
+							innerBlocks.push(
+								createBlock( 'core/list', {
+									ordered: false,
+									values: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'OL':
+							innerBlocks.push(
+								createBlock( 'core/list', {
+									ordered: true,
+									values: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'PRE':
+							innerBlocks.push(
+								createBlock( 'core/code', {
+									content: childNode.innerHTML,
+								} )
+							);
+							break;
+						case 'CITE':
+							cite = childNode.innerHTML;
+							break;
+						default:
+							break;
+					}
+				} );
+				return createBlock(
+					'core/quote',
+					{ citation: cite },
+					innerBlocks
```

Comment:
@nosolosw The best thing here would be to use the [rawHandler](https://github.com/WordPress/gutenberg/blob/703e324ac4300f7587e240ce1c710b7f8d23465e/packages/blocks/src/api/raw-handling/index.js#L99) and pass the result as inner blocks.

```js
rawHandler( {
 	HTML: node.innerHTML,
 	mode: 'BLOCKS',
 } );
```